simulation/utils/generate_map.py

```diff
@@ -0,0 +1,172 @@
+import random
+import math
+import matplotlib.pyplot as plt
+from enum import Enum
+import numpy as np
+from scipy.interpolate import BSpline, splprep, splev
+import copy
+
+
+class Difficulty(Enum):
+    """Enum for Difficulty level"""
+    PATHETIC = 0
+    EASY = 1
+    MEDIUM = 2
+    HARD = 3
+    REALLY_HARD = 4
+    WTF = 5
+
+
+def intersect(A, B, C, D):
+    """Helper of is_crossing function
+    Returns True if line segments AB and CD intersect"""
+    ccw = lambda A, B, C: (C[1] - A[1]) * (B[0] - A[0]) > (B[1] - A[1]) * (C[0] - A[0])
+    return ccw(A, C, D) != ccw(B, C, D) and ccw(A, B, C) != ccw(A, B, D)
+
+
+def is_crossing(X, Y, i, j):
+    """Function that checks if two lines ((X[i],Y[i]), (X[i+1], Y[i+1])), ((X[j],Y[j]), (X[j+1],Y[j+1])) intersect"""
+    if i == j:
+        return False
+    try:
+        return intersect((X[i], Y[i]), (X[i + 1], Y[i + 1]), (X[j], Y[j]), (X[j + 1], Y[j + 1]))
```

Comment:
Może przypisać to do jakiś tuple'i line_one, line_two wcześniej w tej metodzie? Wtedy operacja intersect będzie wyglądać intersect(line_one, line_two), a rozszyfrowanie jak wygląda każda linia będzie łatwiejsze, bo kilka linijek wyżej będzie wyraźne line_one((X[i],Y[i]), (X[i+1], Y[i+1])).